CSF Factories: Web Components args should allow kebab-case attribute names

## Describe the bug

When using CSF factories with web components and libraries like `@wc-toolkit/storybook-helpers`, TypeScript errors occur because args use kebab-case attribute names (e.g., `'static-color'`) while our types expect camelCase property names (e.g., `staticColor`).

## Example

```ts
// @wc-toolkit/storybook-helpers generates args with kebab-case:
const meta = preview.meta({
    component: 'swc-divider',
    render: (args) => template(args),
});

export const StaticBlack = meta.story({
    args: {
        'static-color': 'black',  // ❌ TS Error: 'static-color' does not exist in type
    },
});
```

## Root cause

- `@wc-toolkit/storybook-helpers` reads the Custom Elements Manifest which has both:
  - `name: "static-color"` (kebab-case attribute)
  - `fieldName: "staticColor"` (camelCase property)
- They use the attribute name (kebab-case) for args keys
- CSF factories infers args from `Partial<Component>` which uses camelCase property names
- At compile time, we don't have access to the manifest to know the attribute names

## Proposed solution

Allow any string containing a dash (`-`) in the args type:

```ts
type Args<T> = Partial<T> & Record<`${string}-${string}`, unknown>
```

This works because:
- JavaScript property names cannot contain `-` (syntax error)
- So anything with `-` is definitively an HTML attribute name
- camelCase props → still type checked ✅
- kebab-case attrs → allowed ✅
- typos without dash → still caught as error ✅

## Reproduction

**Repo:** https://github.com/kasperpeulen/spectrum-web-components/tree/storybook-csf-factories-kebab-case-issue

To reproduce:
```bash
git clone https://github.com/kasperpeulen/spectrum-web-components.git
cd spectrum-web-components
git checkout storybook-csf-factories-kebab-case-issue
yarn install
cd 2nd-gen/packages/swc
npx tsc --noEmit
```

You'll see errors like:
```
components/badge/stories/badge.stories.ts: error TS2769: 
  Object literal may only specify known properties, and '['icon-slot']' does not exist in type
components/divider/stories/divider.stories.ts: error TS2769:
  Object literal may only specify known properties, and ''static-color'' does not exist in type
```

## Environment

- Storybook version: 10.x (CSF factories)
- Framework: `@storybook/web-components-vite`
- Related library: `@wc-toolkit/storybook-helpers`
- Test repo: [Adobe Spectrum Web Components](https://github.com/adobe/spectrum-web-components)

--- a/.gitignore
+++ b/.gitignore
@@ -75,3 +75,4 @@ code/core/report
 
 .junie
 CLAUDE.local.md
+.claude

--- a/code/renderers/web-components/src/csf-factories.test.ts
+++ b/code/renderers/web-components/src/csf-factories.test.ts
@@ -205,3 +205,23 @@ it('Components without Props can be used', () => {
 
   const Basic = meta.story();
 });
+
+// https://github.com/storybookjs/storybook/issues/33524
+it('✅ Kebab-case HTML attribute names are allowed in args', () => {
+  const meta = preview.meta({
+    component: 'my-button',
+    args: {
+      label: 'hello',
+      'aria-label': 'my button', // kebab-case attribute
+    },
+  });
+
+  const Basic = meta.story({
+    args: {
+      'data-testid': 'button-1', // kebab-case attribute
+    },
+  });
+
+  expect(meta.input.args?.['aria-label']).toBe('my button');
+  expect(Basic.input.args?.['data-testid']).toBe('button-1');
+});

--- a/code/renderers/web-components/src/preview.ts
+++ b/code/renderers/web-components/src/preview.ts
@@ -60,6 +60,15 @@ type InferWebComponentsTypes<T, TArgs, Decorators> = WebComponentsTypes &
   T & { args: Simplify<InferArgs<TArgs, T, Decorators>> };
 
 /**
+ * Infers args from a web component's HTMLElement type, allowing both camelCase properties and
+ * kebab-case HTML attribute names (e.g., 'aria-label', 'data-testid', 'static-color').
+ */
+type InferArgsFromComponent<C extends keyof HTMLElementTagNameMap> = Partial<
+  HTMLElementTagNameMap[C]
+> &
+  Record<`${string}-${string}`, unknown>;
+
+/**
  * Web Components-specific Preview interface that provides type-safe CSF factory methods.
  *
  * Use `preview.meta()` to create a meta configuration for a component, and then `meta.story()` to
@@ -94,22 +103,20 @@ export interface WebComponentsPreview<T extends AddonTypes> extends Preview<
     C extends keyof HTMLElementTagNameMap,
     Decorators extends DecoratorFunction<WebComponentsTypes & T, any>,
     // Try to make Exact<Partial<TArgs>, TMetaArgs> work
-    TMetaArgs extends Partial<HTMLElementTagNameMap[C] & T['args']>,
+    TMetaArgs extends InferArgsFromComponent<C> & Partial<T['args']>,
   >(
     meta: {
       component?: C;
       args?: TMetaArgs;
       decorators?: Decorators | Decorators[];
     } & Omit<
-      ComponentAnnotations<WebComponentsTypes & T, Partial<HTMLElementTagNameMap[C]> & T['args']>,
+      ComponentAnnotations<WebComponentsTypes & T, InferArgsFromComponent<C> & T['args']>,
       'decorators' | 'component' | 'args'
     >
   ): WebComponentsMeta<
-    InferWebComponentsTypes<T, Partial<HTMLElementTagNameMap[C]>, Decorators>,
+    InferWebComponentsTypes<T, InferArgsFromComponent<C>, Decorators>,
     Omit<
-      ComponentAnnotations<
-        InferWebComponentsTypes<T, Partial<HTMLElementTagNameMap[C]>, Decorators>
-      >,
+      ComponentAnnotations<InferWebComponentsTypes<T, InferArgsFromComponent<C>, Decorators>>,
       'args'
     > & {
       args: {} extends TMetaArgs ? {} : TMetaArgs;

